rbd/rbd_sparsify.go

```diff
@@ -0,0 +1,82 @@
+//go:build !nautilus && ceph_preview
+// +build !nautilus,ceph_preview
+
+package rbd
+
+/*
+#cgo LDFLAGS: -lrbd
+#include <errno.h>
+#include <stdlib.h>
+#include <rbd/librbd.h>
+
+extern int sparsifyCallback(uint64_t, uint64_t, uintptr_t);
+
+// inline wrapper to cast uintptr_t to void*
+static inline int wrap_rbd_sparsify_with_progress(
+		rbd_image_t image, size_t sparse_size, uintptr_t arg) {
+	return rbd_sparsify_with_progress(
+		image, sparse_size, (librbd_progress_fn_t)sparsifyCallback, (void*)arg);
+};
+*/
+import "C"
+
+import (
+	"github.com/ceph/go-ceph/internal/callbacks"
+)
+
+// SparsifyCallback defines the function signature needed for the
```

Comment:
We should document the semantics of the callback parameters here. (Or is it something super obvious that I don't get?) That would also include the return value, so better we document it here as well, instead of in the docs for `SparsifyWithProgress`.

Reply:
When you say semantics, you mean an `implementation` line similar to the other function doc comments, right?
Or do you mean something else?

Reply:
@Nikhil-Ladha I just meant that I - as a user of this library - would have no idea what the parameters of the callback mean. On the other hand, you are right that we often just refer to the functions of the C API of the ceph libraries, which is also not very user-friendly, because I know how hard it is sometimes, to find the right documentation. But if it is worth mentioning the behavior of the return value, it should also be worth to document the semantics of the parameters. Actually I'm not sure if we have a clear pattern of what to document and what to refer to the upstream docs. @phlogistonjohn, you have an opinion or clearer rule in your mind?

Reply:
I think in a case like this it is good to encourage putting in some docs that explain what the expected parameters are. Note that we didn't do it for group_snap.go, so I wouldn't demand it mysefl. That said,  I think that in this case it is fairly straightforward to add something based on the names of the argument to the librbd_progress_fn_t  (at https://github.com/ceph/ceph/blob/2dee65108fc1e4adfad36ac9bcb4fd35dd8f6420/src/include/rbd/librbd.h#L77 ). So something like `The callback function will be called with the first argument containing the current offset within the image being made sparse and the second argument containing the total size of the image. The third argument is an opaque value that is passed to the SparsifyWithProgress function's data argument and every call to the callback will receive the same object.`  Feel free to take, leave, or modify to your liking.